z3c/dependencychecker/tests/test_imports_database.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+from z3c.dependencychecker.dotted_name import DottedName
+from z3c.dependencychecker.db import ImportsDatabase
+
+
+def test_no_dependencies():
+    database = ImportsDatabase()
+    assert database.get_requirements_names() == []
+
+
+def test_add_dependencies():
+    database = ImportsDatabase()
+    database.set_requirements([
+        DottedName('one'),
+        DottedName('two'),
+    ])
+
+    names = database.get_requirements_names()
+    assert len(names) == 2
+    assert 'one' in names
+    assert 'two' in names
+
+
+def test_declared_dependencies_twice_warning(caplog):
+    database = ImportsDatabase()
+    database.set_requirements([
+        DottedName('one'),
+        DottedName('two'),
+        DottedName('three'),
+        DottedName('one'),
+    ])
+
+    last_log = caplog.records[-1]
+    assert '"one" is declared twice on setup.py' == last_log.message
+
+
+def test_declared_dependencies_no_duplicates():
```

Comment:
You're testing without duplicates? I see two "one" items in the test.

Reply:
Oh, I see, with this test I meant "check that duplicates are removed", so that there are no duplicates after processing them. I will add a docstring on it.